### PR TITLE
Feat/collecthsmetrics

### PR DESCRIPTION
**Pull request by ramprasadn**
Adds collecthsmetrics from picard to bam QC
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker -stub`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: workflows/raredisease.nf [hunk 1]
```diff
@@ -164,7 +168,7 @@ workflow RAREDISEASE {
         PREPARE_GENOME.out.fasta,
         PREPARE_GENOME.out.fai,
         INPUT_CHECK.out.ch_case_info,
-        ch_target_bed
+        ch_target_bed.bed
```
Comment (jemten) [inline workflows/raredisease.nf:171]: shouldn't this be `ch_target_bed.out.bed`? I guess that you have tested it though and knows that it shouldn't 😅

**Review by jemten — COMMENTED** [review]
Good work. Was a bit confused by the cat_cat thing but I believe I got it know. Let me know what you think about publishing processed references in a separate output folder 😄

File: conf/modules.config [hunk 2]
```diff
@@ -73,13 +73,42 @@ process {
...
+    withName: 'TABIX_PT|TABIX_PBT' {
+        publishDir = [
+            path: { "${params.outdir}/bed_check" },
+            mode: 'copy',
+            saveAs: { filename -> filename.equals('versions.yml') ? null : filename }
+        ]
+    }
+    withName: 'GATK_BILT' {
+        publishDir = [
+            path: { "${params.outdir}/bed_check" },
```
[the remark this answers is not in the document]
Comment (projectoriented) [reply conf/modules.config:92]: Yup! I agree here

File: workflows/raredisease.nf (hunk 1 above)
Comment (projectoriented) [reply workflows/raredisease.nf:171]: Yeah but actually I'm curious why it doesn't have to include the `.out`

File: conf/modules.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (projectoriented) [reply conf/modules.config:92]: Yup! I agree here ^

File: workflows/raredisease.nf (hunk 1 above)
Comment (projectoriented) [reply workflows/raredisease.nf:171]: Yeah but actually I'm curious why it doesn't have to include `.out`

File: conf/modules.config (hunk 2 above)
[the remark this answers is not in the document]
Comment (ramprasadn) [reply conf/modules.config:92]: That's a good point! I will fix that 👍🏻

File: workflows/raredisease.nf (hunk 1 above)
Comment (ramprasadn) [reply workflows/raredisease.nf:171]: What I have done in this case is that, I have assigned the output from the CHECK_BED subworkflow i.e CHECK_BED.out to ch_target_bed (line 119). So now when I need to access the attributes from the subworkflow I can simply use ch_target_bed.attribute. Or in other words, ch_target_bed.bed is the same as writing CHECK_BED.out.bed.

**Review by jemten — APPROVED** [review]
🎉

**Comment by projectoriented** [comment]
Issue not referenced, this closes https://github.com/nf-core/raredisease/issues/42

**Pull request merged**